tools/shared.py

```diff
@@ -1950,7 +1950,7 @@ def llvm_opt(filename, opts, out=None):
 
     logging.debug('emcc: LLVM opts: ' + ' '.join(opts) + '  [num inputs: ' + str(len(inputs)) + ']')
     target = out or (filename + '.opt.bc')
-    proc = run_process([LLVM_OPT] + inputs + opts + ['-o', target], stdout=PIPE)
+    proc = run_process([LLVM_OPT] + inputs + opts + ['-o', target], stdout=PIPE, check=False)
```

Comment:
as I wrote in the other PR, I don't think this is good. this is a very central location, all our invocations of llvm-opt go through here, so it would be good to keep checking for errors here, and it seems this is only done to fix a single test's issue - can't we fix the test? maybe I am missing something here.

Reply:
We do test for errors here a couple of lines down already, there's an explicit check for `if proc.returncode != 0` condition. The refactor here had made that the `proc.returncode != 0` case would never be reached - I think the correct fix is to add `check=False` to allow returncode != 0 to propagate back to here so that the code can handle it below. Or if we don't want to do that anymore, then we should delete the `if proc.returncode != 0` case below, since it can no longer be reached.

The test is for a negative failure case, ensuring that a custom error message is printed when attempting to link incorrect native linked libs. The generic failure caused that this helping error message was no longer being shown.

Reply:
Oh, I see now, sorry for missing that. Yes, that makes sense. lgtm!